### DISCUSSION TITLE
Wrap attributes

### DISCUSSION THREAD

**Discussion by CheIIau** [vuejs/language-tools#988] [Q&A] [answered]
Maybe it's a stupid question but i can't figure it out. Can anybody help me how to wrap attibutes per line using volar as a formatter?
I want this template...
![image](https://user-images.githubusercontent.com/58114735/155850328-acbe532b-d3d3-4a9b-90fc-36903940d9ae.png)
...to be formatted to this.
![image](https://user-images.githubusercontent.com/58114735/155850387-ee0b08fd-7d3f-4773-84f3-86a691a7e8c4.png)
In vetur it was easy to do. In volar i can't find a solution

**Comment by GIBZ-GIPE** [comment]
I have exactly the same problem as @CheIIau: I can't align attributes of neither native HTML elements nor Vue components as seen in the official docs.

My configuration for Visual Studio Code is:

```json
  "html.format.wrapAttributes": "force-expand-multiline",
  "[vue]": {
    "editor.defaultFormatter": "johnsoncodehk.volar",
    "editor.formatOnSave": true,
    "editor.formatOnSaveMode": "file",
  },
  "workbench.editor.splitInGroupLayout": "vertical"
```

**Comment by johnsoncodehk** [comment]
You could try given a smaller `volar.formatting.printWidth` in settings.

**Reply by GIBZ-GIPE** [reply]
Thanks, that helped 👍
Although, I'm not really sure I do understand. I expect `volar.formatting.printWidth` to relate to the general "length of a line", not only to attribute wrapping. Are there any side effects to be aware of?

**Comment by maqingbo** [comment, marked as answer]
Maybe you need this in `setting.json`:

```json
"html.format.wrapAttributes": "force-aligned"
```

**Reply by fatihaziz** [reply]
It's the correct answer! thank you it help!

**Reply by MoWafa** [reply]
Thanks @maqingbo, this is the correct answer.

**Reply by and3rson** [reply]
Sorry for necrobumping, but how do I configure this with nvim-lspconfig? E. g. here's my config:

```lua
lspconfig.volar.setup {
  on_attach = on_attach,
  filetypes = { 'typescript', 'javascript', 'javascriptreact', 'typescriptreact', 'vue' },
  init_options = {
    vue = {
      hybridMode = false,
    },
    typescript = {
        tsdk = "/usr/lib/node_modules/typescript/lib/",
    },
  },
}
```

- How do I add `html.format.wrapAttributes` in there?
- How do I force `html.format.wrapAttributes` on a per-project basis, so that other devs also use the same alignment with their formatters?

### Edit
I figured it out:

```lua
lspconfig.volar.setup {
  on_attach = on_attach,
  filetypes = { 'typescript', 'javascript', 'javascriptreact', 'typescriptreact', 'vue' },
  init_options = {
    vue = {
      hybridMode = false,
    },
    typescript = {
        tsdk = "/usr/lib/node_modules/typescript/lib/",
    },
  },
  settings = {
    html = {
      format = {
        wrapAttributes = "force-expand-multiline",
      },
    },
  },
}
```

Now I wish there was a way to configure this on a per-project basis.

**Comment by joantp** [comment]
may be it is a stupid question ... but ... Can anybody help me how to **avoid**, **prevent volar from** wrapping attibutes per line using volar as a formatter?
Thank you!

**Reply by logan2d5** [reply]
![image](https://github.com/user-attachments/assets/04bb9844-6f27-4a23-aba5-9c7033ab718c)

Reference: [HTML Programming with Visual Studio Code](https://code.visualstudio.com/docs/languages/html#_formatting)

**Reply by kshksdrt** [reply]
There is a prettier configuration and also an eslint rule for that
